tests/BuiltInWebServerAbstract.php

```diff
@@ -109,9 +118,87 @@ private static function getPackagePath($directory = null, $_ = null): string
 
     private function getClient(): Client
     {
+        if (static::isWithWebserverTest()) {
+            $handler = null;
+        } else {
+            $handler = function (\Psr\Http\Message\RequestInterface $request, array $options) {
+                try {
+                    // backup globals
+                    $bak = [];
+                    foreach ([
```

Comment:
it will be not bad to add `laminas/laminas-diactoros` and set it as an injectable property of the App, and remove all direct access to superglobal, it can be a first step in direction of PSR-7.
```php
ServerRequestFactory::fromGlobals(
            $_SERVER,
            $_GET,
            $_POST,
            $_COOKIE,
            $_FILES
        );
```